OpenRA.Game/Activities/Activity.cs

```diff
@@ -31,45 +31,56 @@ public abstract class Activity
 	{
 		public ActivityState State { get; private set; }
 
-		Activity childActivity;
-		protected Activity ChildActivity
-		{
-			get { return childActivity != null && childActivity.State < ActivityState.Done ? childActivity : null; }
-			set { childActivity = value; }
-		}
-
+		protected Activity ChildActivity { get; private set; }
 		public Activity NextActivity { get; protected set; }
 
 		public bool IsInterruptible { get; protected set; }
+		public bool ChildHasPriority { get; protected set; }
 		public bool IsCanceling { get { return State == ActivityState.Canceling; } }
 
 		public Activity()
 		{
 			IsInterruptible = true;
+			ChildHasPriority = true;
 		}
 
 		public Activity TickOuter(Actor self)
 		{
-			if (State == ActivityState.Done && Game.Settings.Debug.StrictActivityChecking)
```

Comment:
This was the only place that used `StrictActivityChecking`, so that can now be removed from `DebugSettings` and the advanced settings UI.

Reply:
Done.